Public/Src/Cache/ContentStore/Distributed/Sessions/DistributedContentSession.cs

```diff
@@ -165,5 +180,51 @@ private async Task<PutResult> RegisterPutAsync(Context context, CancellationToke
 
             return putResult;
         }
+
+        private async Task<BoolResult> RequestProactiveCopyIfNeededAsync(OperationContext context, ContentHash hash)
+        {
+            var hashArray = new[] { hash };
+
+            // First check in local location store, then global if failed.
+            var getLocationsResult = await ContentLocationStore.GetBulkAsync(context, hashArray, context.Token, UrgencyHint.Nominal, GetBulkOrigin.Local);
+            if (!getLocationsResult.Succeeded || getLocationsResult.ContentHashesInfo[0].Locations.Count == 0)

```

Comment:
Is it possible that ContentHashesInfo is empty?

Reply:
If it is, it is a bug and we're right to blow up, IMO

Reply:
If it is, it is a bug and we're right to blow up, IMO

---
In reply to: [312225497](https://github.com/microsoft/BuildXL/pull/705#discussion_r312225497) [](ancestors = 312225497)